src/data/feature.puml

```diff
@@ -9,25 +12,35 @@ BorderColor black
 FontName Segoe UI
 FontSize 15
 }
-(*) --> "Read ?"
---> "AST"
---> "handle Feature"
+(*) -->[Feature] "handle Feature"
 -right-> "onFeature"
--left-> "handle Feature"
---> "filter/pre Tags"
--right-> "prefilter Tag"
+-left-> "handle Feature"<<handle>>
+--> "filter/pre Tags"<<filter>>
+-right-> "pre Tags"
 -left-> "filter/pre Tags"
---> "handle tag"
--right-> "on Tag"
--left-> "handle tag"
---> "filter/post Tags"
--right-> "postfilter Tag"
+--> "Apply Tags"<<apply>>
+--> "filter/post Tags"<<filter>>
+-right-> "post Tags"
 -left-> "filter/post Tags"
---> "Process Elements"
-If "rule?"
---> [Yes] "applyRule"
---> "applyElements"
+--> "filter/pre Rule"<<filter>>
+-right-> "pre Rule"
+-left-> "filter/pre Rule"
+--> "Apply Rule"<<apply>>
+If "Are there rules left?"
+-left->[Yes] "Apply Rule"
 else
---> [No] "applyElements"
+--> [No] "filter/post Rule"
+-right-> "post Rule"
+-left-> "filter/post Rule"<<filter>>
+--> "filter/pre Scenario"<<filter>>
```

Comment:
Handling the rule and the normal elements are mutually exclusive so you either have rules in the feature and you process it, or you have elements (Scenario, Background, etc) and process those, but not the same time.